x/gov/client/cli/tx.go

```diff
@@ -135,6 +136,40 @@ Where proposal.json contains:
 	return cmd
 }
 
+// NewCmdCancelProposal implements submitting a cancel proposal transaction command.
+func NewCmdCancelProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "cancel-proposal [proposal-id]",
+		Short: "Cancel governance proposal by proposal before the voting period ends.",
+		Args:  cobra.ExactArgs(1),
+		Example: fmt.Sprintf(`
+		$ %s tx gov cancel-proposal 1 --from mykey
+		`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			// validate that the proposal id is a uint
+			proposalID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("proposal-id %s not a valid uint, please input a valid proposal-id", args[0])
+			}
+
+			// Get depositor address
```

Comment:
This should be the proposer address right?
```suggestion
			// Get proposer address
```